account/models.py

```diff
@@ -72,6 +76,27 @@ class CustomUser(AbstractBaseUser, PermissionsMixin):
 
     def __str__(self):
         return self.email
+    
+    def clean(self):
+        super().clean()
+        
+        # Validate gov_id length based on gov_id_prefix
+        prefix = self.gov_id_prefix
+        gov_id_length = len(self.gov_id)
+        
+        if prefix == 'AZE' and gov_id_length != 8:
```

Comment:
You have to check if everything is digit, so use regex please

Reply:
You can use dict
`validators = {"AZE": regex1, "AA": regex2}`
And then, 
`if re.match(gov_id, validators[pin_code])`
To reduce if block size

Reply:
I didnt get the way how this will reduce if block size. But I came out with cleaner code using regex.
        
       regex1 = '^\d{8}$'
        regex2 = ['^\d{7}$']
        regex3 = '^(?:\d{5}|\d{6})$'
        validators = {'AZE': regex1, 'AA': regex2, 'MYI': regex3, 'DYI': regex3}
        prefix = self.gov_id_prefix
       
        if re.match('^[0-9]+$', self.gov_id):
            if prefix == 'AZE' and not re.match(validators['AZE'], self.gov_id):
                raise ValidationError("The gov_id field length must be 8 for AZE prefix.")
            elif prefix == 'AA' and not re.match(validators['AA'], self.gov_id):
                raise ValidationError("The gov_id field length must be 7 for AA prefix.")
            elif prefix == 'MYI' and not re.match(validators['MYI'], self.gov_id):
                raise ValidationError("The gov_id field length must be 5 or 6 for MYI prefix.")
            elif prefix == 'DYI' and not re.match(validators['DYI'], self.gov_id):
                raise ValidationError("The gov_id field length must be 5 or 6 for DYI prefix.")
        else:
            raise ValidationError("All numbers have to be digit")

Reply:
why do you write if prefix == AZE? Just do re.match(validators[prefix], self.gov_id), you dont need 4 if/elseifs

Reply:
I think only reason is to give specific validationerror message for each gov id prefix. But right I can just ask for right number of digits.